core/src/main/java/com/linecorp/armeria/server/ServerHttp1ObjectEncoder.java

```diff
@@ -106,6 +108,8 @@ private HttpResponse convertHeaders(ResponseHeaders headers, boolean endStream,
 
             if (HttpStatus.isContentAlwaysEmpty(statusCode)) {
                 maybeRemoveContentLength(statusCode, outHeaders);
+            } else if (method == HttpMethod.HEAD) {
+                maybeSetTransferEncoding(res, statusCode);
```

Comment:
Could you explain why we need to set transfer encoding headers for a response to a HEAD request? I thought we can just avoid writing `content-length` and that should be just enough.

According to the section 4.3.2 of RFC 7231:

> The server SHOULD send the same
> header fields in response to a HEAD request as it would have sent if
> the request had been a GET, except that the payload header fields
> ([Section 3.3](https://www.rfc-editor.org/rfc/rfc7231#section-3.3)) MAY be omitted.

Reply:
I just wanted to avoid writing `0` length, an incorrect value, for transfer encoding. It sounds good to skip writing for transfer encoding.